backend/database/auth.py

```
@@ -1,7 +1,43 @@
+    """Update user document"""
+    db = get_firestore()
+    user_ref = db.collection('users').document(uid)
+    user_ref.update(user_data)
+
+def validate_token(token: str) -> str:
+    """Validate Firebase ID token"""
+    try:
+        decoded_token = auth.verify_id_token(token)
+        return decoded_token['uid']
+    except Exception as e:
+        raise HTTPException(status_code=401, detail=str(e))
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)